subprojects/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/ApiGroovyCompiler.java

```diff
@@ -253,4 +277,51 @@ private void canonicalizeValues(Map<String, Boolean> options) {
     private ClassLoader getExtClassLoader() {
         return ClassLoaderUtils.getPlatformClassLoader();
     }
+
+    private static class TrackingCompilationCustomizer extends CompilationCustomizer {
+        public TrackingCompilationCustomizer() {
+            super(CompilePhase.CLASS_GENERATION);
+        }
+
+        private final Map<URI, List<String>> location2ClassNames = new HashMap<URI, List<String>>();
+
+        @Override
+        public void call(SourceUnit source, GeneratorContext context, ClassNode classNode) throws org.codehaus.groovy.control.CompilationFailedException {
+            inspectClassNode(source, classNode);
+        }
+
+        void inspectClassNode(SourceUnit sourceUnit, ClassNode classNode) {
+            location2ClassNames.computeIfAbsent(sourceUnit.getSource().getURI(), new Function<URI, List<String>>() {
```

Comment:
`computeIfAbsent`/`java.util.function.Fuction` are Java 8 API but this module is marked as `ModuleType.WORKER`. Is this accidental or intentional?